components/stepper/index.css

```
@@ -10,134 +10,560 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+    --highcontrast-stepper-border-color-invalid-keyboard-focus: Highlight;
+
+    --highcontrast-stepper-button-background-color-default: Canvas;
+    --highcontrast-stepper-button-background-color-hover: Canvas;
+    --highcontrast-stepper-button-background-color-focus: Canvas;
+    --highcontrast-stepper-button-background-color-keyboard-focus: Canvas;
+
+    --highcontrast-stepper-focus-indicator-color: CanvasText;
+  }
+}
+.x {
+ border-radius: var(--spectrum-stepper-button-border-radius-reset);
 }
 
 .spectrum-Stepper {
+  position: relative;
   display: inline-flex;
   flex-direction: row;
   flex-wrap: nowrap;
 
-  inline-size: var(--spectrum-stepper-width);
+  inline-size: var(--mod-stepper-width, var(--spectrum-stepper-width));
   line-height: 0;
-  border-radius: var(--spectrum-alias-border-radius-regular);
-  transition: border-color var(--spectrum-global-animation-duration-100) ease-in-out, box-shadow var(--spectrum-global-animation-duration-100) ease-in-out;
+  border-radius: var(--mod-stepper-border-radius, var(--spectrum-stepper-border-radius)); /* --spectrum-global-dimension-size-75 = 6px 8px */
+
+  /*** Focus Indicator - Unfocused Base Styles ***/
+  &::after {
+    block-size: calc(100% + (var(--mod-stepper-focus-indicator-gap, var(--spectrum-stepper-focus-indicator-gap)) * 2));
+    inline-size: calc(100% + (var(--mod-stepper-focus-indicator-gap, var(--spectrum-stepper-focus-indicator-gap)) * 2));
+    position: absolute;
+    left: 0;
+    right: 0;
+    top: 0;
+    pointer-events: none;
+    content: '';
+    border-style: solid;
+    border-color: transparent;
+    border-width: 0;
+    border-radius: calc(var(--mod-stepper-border-radius, var(--spectrum-stepper-border-radius))
+                        + var(--mod-stepper-border-width, var(--spectrum-stepper-border-width))
+                        + var(--mod-stepper-focus-indicator-gap, var(--spectrum-stepper-focus-indicator-gap)));
+
+    margin-block-start: calc((var(--mod-stepper-focus-indicator-gap, var(--spectrum-stepper-focus-indicator-gap))
+                              + var(--mod-stepper-focus-indicator-width, var(--spectrum-stepper-focus-indicator-width)))
+                              * -1);
+    margin-inline-start: calc((var(--mod-stepper-focus-indicator-gap, var(--spectrum-stepper-focus-indicator-gap))
+                              + var(--mod-stepper-focus-indicator-width, var(--spectrum-stepper-focus-indicator-width)))
+                              * -1);
+  }
+
+  /*** Hover ***/
+  &:hover:not(.is-disabled):not(.is-invalid):not(.is-focused):not(.is-keyboardFocused) {
+    .spectrum-Stepper-buttons,
+    .spectrum-Stepper-stepUp,
+    .spectrum-Stepper-stepDown,
+    .spectrum-Stepper-input {
+      border-color: var(--highcontrast-stepper-border-color-hover, var(--mod-stepper-border-color-hover, var(--spectrum-stepper-border-color-hover)));
+    }
+  }
+
+
+  /*** Focused ***/
+  &.is-focused {
```

```suggestion
  &.is-focused,
  &:focus {
```

I wonder if it makes sense to include the class as well as the pseudo selector so that the class only becomes necessary if you need to apply focus manually.

---

Ok so after looking at this more, I think we may need to force people to use the classes. There are other things depending on those classes which you can see below this line
```
:hover {
        .spectrum-Stepper-input,
        .spectrum-Stepper-buttons,
        .spectrum-Stepper-stepUp,
        .spectrum-Stepper-stepDown
```
 that depend on the parent class for styling. I can add additional CSS for each one individually for `focus`, `disabled`, etc. but I feel like that may just make things confusing. What do you think?

---

I don't know if `:has(input:focus)` would work but I could try that